### PR TITLE
feat(devtools): improve draggable action button & route tester

### PR DESCRIPTION
**Pull request by Asuka109**
### <samp>🤖 Generated by Copilot at b9fb4ae</samp>

This pull request refactors and improves the devtools package and its client, plugin, and kit modules:

- Removes unused and hardcoded values.
- Fixes URL handling and layout issues.
- Adds support for custom tabs and a draggable action button.
- Reorganizes the types and constants.

It also updates the devtools route prefix to use a constant from `@modern-js/devtools-kit` and fixes some test and type declaration issues.

### <samp>🤖 Generated by Copilot at b9fb4ae</samp>

*  Add support for custom tabs in the devtools UI that can render an iframe with a given source ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-04521cf71e80dda8987945a0de3b1063d8d3f1e8027666b67e5b084b3f248523R42-R55), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-789b57807392e9f547a9d41ee50684b3336e468226238d006d5267802e70070bL1-R5))
*  Use a constant for the devtools route prefix instead of hardcoding it ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-249610e4c3d6bd8dbb483f2667c5093d38461da73b5a217a63a55cbe74e0f2a3L2-R2), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-249610e4c3d6bd8dbb483f2667c5093d38461da73b5a217a63a55cbe74e0f2a3L9-R13), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-27ca05cbb672278c26c774086ebe4d0c2de885b1a991a33493f84b38bf86c2b6R1), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-4181b47f031e7472635d656365b8e26222b95ad813741a30f04f8d7446ca9152L15-R19), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-2f846ce94e6453db4870e9758df4c3b79d2bb5031b231559a139fbb549b54655L7-R11), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-2f846ce94e6453db4870e9758df4c3b79d2bb5031b231559a139fbb549b54655L63-R68), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-2f846ce94e6453db4870e9758df4c3b79d2bb5031b231559a139fbb549b54655L84-R91))
*  Use a custom hook for making the devtools action button draggable and sticky to the edges of the window ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-4181b47f031e7472635d656365b8e26222b95ad813741a30f04f8d7446ca9152L1-R8), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-4181b47f031e7472635d656365b8e26222b95ad813741a30f04f8d7446ca9152L83-R51), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-01a4360517dda7eb48554d0c0ee77ec9d7e8a22ddcfb7dad7cb0750bda9d54f1R1-R145))
*  Use a more robust and consistent way of joining and normalizing URLs using `ufo` ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-f6109168e26bc3d8c885cae7971b76050875f5b720cbdcc6d07c391a4c670771L6-R6), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-f6109168e26bc3d8c885cae7971b76050875f5b720cbdcc6d07c391a4c670771L21-R22), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-6eef33196afd7166dd1a670423d6839d4fbc3ebf3e0e942b591435119f2ee6b0R9), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-6eef33196afd7166dd1a670423d6839d4fbc3ebf3e0e942b591435119f2ee6b0L25-R26), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-6eef33196afd7166dd1a670423d6839d4fbc3ebf3e0e942b591435119f2ee6b0L38-R39))
*  Fix layout issues where the client and server routes would overflow the container and cause unwanted scrolling ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-b21602326b4ff8ea3bd1ecdc88bdee3ee99557bcb96f451171ea66179c447654L2-R2), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-b21602326b4ff8ea3bd1ecdc88bdee3ee99557bcb96f451171ea66179c447654L17-R19), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-956b3da3a2b3070e26d75a3ff759b3de73a6ce202642fc82ccfc8dccdd65e85bL35-R41), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-956b3da3a2b3070e26d75a3ff759b3de73a6ce202642fc82ccfc8dccdd65e85bL47-R59))
*  Fix a bug where the root index route would not show the `EndpointTag` ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-f6109168e26bc3d8c885cae7971b76050875f5b720cbdcc6d07c391a4c670771L38-R36))
*  Simplify the config and remove unused options and dependencies ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-249610e4c3d6bd8dbb483f2667c5093d38461da73b5a217a63a55cbe74e0f2a3L23-R26), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-5e1ff452f8374a96ed9e1e0479f7b0262e971402054d29c88ab55833b2c9f307L7), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-5e1ff452f8374a96ed9e1e0479f7b0262e971402054d29c88ab55833b2c9f307L13))
*  Fix a typo in the path to the type declaration file ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-2dfba75a589b54f8df7db5664a8362f013d0f9988e00826b5119edd12d7ba65cL88-R88))
*  Fix a test issue where the main entry name was not recognized by the bundler ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-eefad722307332d0f286382a1a9885208fede4ee072192edb7ff9dafbe23053eR8-R10))
*  Reorganize the types and exports and remove unused files ([link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-04521cf71e80dda8987945a0de3b1063d8d3f1e8027666b67e5b084b3f248523R42-R55), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-789b57807392e9f547a9d41ee50684b3336e468226238d006d5267802e70070bL1-R5), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-afca80be41e878894e80a4d8de0ae8db6fc2306265319b8c7a2ec92f68a8afe8), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-f5b83bb4c590fe313d4006a40cc1abd030aa44a585d93b35fe2bc3dc3fc11c43L1-L2), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-4ae44db75f319316cb68762ceb3d05866783735f625bc86ef842caccaf0b90ca), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-d98e1aa26172f7c9d1cd90a945debbb43dd40de36ca584c96281aafdcac1d2c9), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-9e086e11180efa3bda32543c6d10c937d79b893def93640feae69f8cd5084497L1-R1), [link](https://github.com/web-infra-dev/modern.js/pull/4721/files?diff=unified&w=0#diff-c3365fadae8acbd432aeea57cc47e611f90e51d206495cfd6a9ae02cefe4f1ea))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: tests/integration/devtools/src/main/routes/page.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { Helmet } from '@modern-js/runtime/head';
+import './index.css';
+
+const Index = (): JSX.Element => (
+  <div className="container-box">
+    <Helmet>
+      <link
+        rel="icon"
+        type="image/x-icon"
+        href="https://lf3-static.bytednsdoc.com/obj/eden-cn/upspbovhj/edenx-ico.ico"
```
Comment (chenjiahan) [inline tests/integration/devtools/src/main/routes/page.tsx:10]: No EdenX, use Modern.js starter template instead.

File: tests/integration/devtools/src/main/routes/post/[id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+const Page: React.FC = () => (
+  <div>
+    <h3>约定式路由</h3>
+    <p>
+      以 routes/ 为约定的入口，EdenX 会自动基于文件系统，生成对应的路由结构。
```
Comment (chenjiahan) [inline tests/integration/devtools/src/main/routes/post/[id]/page.tsx:7]: English first

Comment (Asuka109) [reply tests/integration/devtools/src/main/routes/post/[id]/page.tsx:7]: Already re-wrote the integration demo.

Comment (chenjiahan) [reply tests/integration/devtools/src/main/routes/post/[id]/page.tsx:7]: Should we delete this file?

Comment (Asuka109) [reply tests/integration/devtools/src/main/routes/post/[id]/page.tsx]: I forget that file... fixed now.

**Pull request merged**